clientv3/watch.go

```diff
@@ -49,6 +49,11 @@ type Watcher interface {
 	Close() error
 }
 
+type CreatedNotifier interface {
```

Comment:
Why not add an extra field into `clientv3.watchRequest` that only matters to the client? The proxy can then do `WithCreateNotify` on every watcher it creates and get the same behavior. I think regular clients might find it useful to know the start revision of particular Watchers (but not all) too. The way it is now doesn't fit well with the rest of the client design.